flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobMasterServiceLeadershipRunner.java

```diff
@@ -376,7 +376,7 @@ private void confirmLeadership(
                                     LOG.debug("Confirm leadership {}.", leaderSessionId);
                                     leaderElection.confirmLeadership(leaderSessionId, address);
                                 } else {
-                                    LOG.trace(
+                                    LOG.debug(
                                             "Ignore confirming leadership because the leader {} is no longer valid.",
                                             leaderSessionId);
                                 }
```

Comment:
the entire callback can be replaced by `runIfValidLeader`

Reply:
These codes have been replaced by `runIfValidLeader`.